geomesa-hbase/geomesa-hbase-tools/bin/aws-utils/.aws-bootstrap-actions.sh

```diff
@@ -0,0 +1,138 @@
+#!/usr/bin/env bash
+#
+# Do not run this script manually. It is part of the automated bootstrap process and is not written for end user use.
+#
+# Core Bootstrap Script for GeoMesa on AWS EMR
+#
+# Note this script runs twice. It is initiated by AWS before the cluster is ready. During its lifetime it downloads a
+# copy of the tools distribution and on the master spawns a clone of itself that waits for the cluster to finish
+# setting up before continuing the bootstrap process.
+
+logFile=/tmp/bootstrap.log
+function log() {
+  timeStamp=$(date +"%T")
+  echo "${timeStamp}| ${@}" | tee -a $logFile
+  sudo -H -u ec2-user aws s3 cp $logFile ${CONTAINER}/bootstrap.log
+}
+
+ARGS=($@) # Save the input args so we can pass them to the child.
+VERSION=%%project.version%%
+GM_TOOLS_DIST="%%gmtools.assembly.name%%"
+GM_TOOLS_HOME="/opt/${GM_TOOLS_DIST}"
+CHILD=false
+CID=$(cat /mnt/var/lib/info/job-flow.json | jq .jobFlowId)
+
+while [[ $# -gt 0 ]]; do
+  case $1 in
+    -z|--zeppelin)
+      ZEPPELIN=true
+      shift
+      ;;
+    -j*|--jupyter*)
+      JUPYTER=true
+      JUPYTER_PASSWORD="${1#*=}"
+      if [[ -z "${JUPYTER_PASSWORD}" ]]; then
+        JUPYTER_PASSWORD="geomesa"
+      fi
+      shift
+      ;;
+    -c=*|--container=*)
+      CONTAINER="${1#*=}"
+      shift
+      ;;
+    --child)
+      CHILD=true
+      shift
+      ;;
+    *)
+      echo "[Warning] Unknown parameter: ${1}"
+      shift
+      ;;
+  esac
+done
+
+log "Bootstrap Actions Spawned"
+
+# Validate Parameters
+if [[ -z "${CONTAINER}" ]]; then
+  log "S3 container is required"
+  exit 1
+elif [[ "${CONTAINER}" != */ ]]; then
+  # We need a trailing '/' for consistency
+  CONTAINER="${CONTAINER}/"
+fi
+
+if [[ "${CHILD}" != "true" ]]; then
+  log "Parent Process"
+
+  # Parses a configuration file put in place by EMR to determine the role of this node
+  isMaster=$(jq '.isMaster' /mnt/var/lib/info/instance.json)
+  log "isMaster: ${isMaster}" 
+
+  ### MAIN ####
+  log "Main" 
+  if [[ "${isMaster}" == "true" ]]; then
+    log "Copying Resources Locally" 
+    sudo aws s3 cp ${CONTAINER}${GM_TOOLS_DIST}-bin.tar.gz /opt/${GM_TOOLS_DIST}-bin.tar.gz
```

Comment:
perhaps should add a check to verify this works? or chain ``&&`` or something?

Reply:
Probably a good idea, the user might not have permissions.